### PR TITLE
Stop and Abort must not complete the step

### PR DESCRIPTION
**Pull request by jluhrs**
There was several side changes. Epics-service (giapi) was updated to provide channels of type Short. Binary libraries were removed from epics.acm. Now they are retrieved from a repository.

### CODE REVIEW

**Comment by cquiroz** [comment]
Could you test deploying this to a test seqexec to make sure the new dependencies are being included correctly?

File: modules/edu.gemini.epics.acm/src/main/java/edu/gemini/epics/acm/CaObserveSenderImpl.java [hunk 1]
```diff
@@ -0,0 +1,701 @@
...
+        }
+        if(abortMark!=null) {
+            try {
+                abortMark.unRegisterListener(abortMarkListener);
+            } catch (CAException e) {
+                LOG.warning(e.getMessage());
+            }
+        }
+
+        apply.unbind();
```
Comment (cquiroz) [inline modules/edu.gemini.epics.acm/src/main/java/edu/gemini/epics/acm/CaObserveSenderImpl.java]: Can unbin throw?

File: modules/edu.gemini.epics.acm/src/main/java/edu/gemini/epics/acm/CaObserveSenderImpl.java [hunk 2]
```diff
@@ -0,0 +1,701 @@
...
+        car.unbind();
+    }
+
+    @Override
+    public synchronized CaCommandMonitor post() {
+        CaCommandMonitorImpl cm = new CaCommandMonitorImpl();
+        if (currentState != IdleState) {
```
Comment (cquiroz) [inline modules/edu.gemini.epics.acm/src/main/java/edu/gemini/epics/acm/CaObserveSenderImpl.java]: Is `IdleStat` an int or primitive? If it is an object you should use `!equals`

File: modules/edu.gemini.epics.acm/src/main/java/edu/gemini/epics/acm/CaObserveSenderImpl.java [hunk 3]
```diff
@@ -0,0 +1,701 @@
...
+
+        @Override
+        public CaObserveSenderImpl.State onTimeout() {
+            failCommand(cm, new TimeoutException());
+            return IdleState;
+        }
+
+        @Override
+        public State onStopMarkChange(Short val) {
+            if(stopMark==2 && val == 0) {
```
Comment (cquiroz) [inline modules/edu.gemini.epics.acm/src/main/java/edu/gemini/epics/acm/CaObserveSenderImpl.java]: Magic 2 number? Perhaps we should use a constant

File: modules/edu.gemini.epics.acm/src/main/java/edu/gemini/epics/acm/CaObserveSenderImpl.java [hunk 4]
```diff
@@ -0,0 +1,701 @@
...
+        this.timeout = timeout;
+        this.timeoutUnit = timeUnit;
+    }
+
+    private void succedCommand(final CaCommandMonitorImpl cm) {
+        executor.execute(() -> cm.completeSuccess());
+    }
+
+    private void pauseCommand(final CaCommandMonitorImpl cm) {
+        executor.execute(() ->cm.completePause());
```
Comment (cquiroz) [inline modules/edu.gemini.epics.acm/src/main/java/edu/gemini/epics/acm/CaObserveSenderImpl.java]: Missing space between `->cm`

File: modules/edu.gemini.epics.acm/src/main/java/edu/gemini/epics/acm/CaObserveSenderImpl.java [hunk 5]
```diff
@@ -0,0 +1,701 @@
...
+
+    private void failCommandWithApplyError(final CaCommandMonitorImpl cm) {
+        // I found that if I try to read OMSS or MESS from the same thread that
+        // is processing a channel notifications, the reads fails with a
+        // timeout. But it works if the read is done later from another thread.
+        executor.execute(() -> {
+            String msg = null;
+            try {
+                msg = apply.getMessValue();
+            } catch (CAException | TimeoutException e) {
```
Comment (cquiroz) [inline modules/edu.gemini.epics.acm/src/main/java/edu/gemini/epics/acm/CaObserveSenderImpl.java]: nice

File: modules/edu.gemini.seqexec.model/src/main/scala/edu/gemini/seqexec/model/ActionType.scala [hunk 6]
```diff
@@ -17,6 +17,7 @@ object ActionType {
   // Used in tests
   object Undefined extends ActionType
   final case class Configure(sys: Resource) extends ActionType
+  
+implicit val equal: Equal[ActionType] = Equal.equalA[ActionType]
```
Comment (cquiroz) [inline modules/edu.gemini.seqexec.model/src/main/scala/edu/gemini/seqexec/model/ActionType.scala]: Missing indentation

File: modules/edu.gemini.seqexec.server/src/main/scala/edu/gemini/seqexec/server/EpicsUtil.scala [hunk 7]
```diff
@@ -77,7 +77,7 @@ trait EpicsSystem[T] {
 @SuppressWarnings(Array("org.wartremover.warts.Overloading"))
 object EpicsCommand {
 
-  trait Result
+  sealed trait Result
```
Comment (cquiroz) [inline modules/edu.gemini.seqexec.server/src/main/scala/edu/gemini/seqexec/server/EpicsUtil.scala]: 👍

File: modules/edu.gemini.seqexec.server/src/main/scala/edu/gemini/seqexec/server/SeqTranslate.scala [hunk 8]
```diff
@@ -43,22 +43,30 @@ class SeqTranslate(site: Site, systems: Systems, settings: Settings) {
   private def dhsFileId(inst: InstrumentSystem): SeqAction[ImageFileId] =
     systems.dhs.createImage(DhsClient.ImageParameters(DhsClient.Permanent, List(inst.contributorName, "dhs-http")))
 
+  private def sendDataStart(obsId: SPObservationID, imageFileId: ImageFileId, dataId: String): SeqAction[Unit] =
+    systems.odb.datasetStart(obsId, dataId, imageFileId).flatMap{
```
Comment (cquiroz) [inline modules/edu.gemini.seqexec.server/src/main/scala/edu/gemini/seqexec/server/SeqTranslate.scala]: I think the `if` could be avoided with something like `foldLeftM` but this is low prio

File: modules/edu.gemini.seqexec.server/src/main/scala/edu/gemini/seqexec/server/SeqTranslate.scala [hunk 9]
```diff
@@ -67,24 +75,38 @@ class SeqTranslate(site: Site, systems: Systems, settings: Settings) {
...
+      //TODO: Implement pause handling
+      case ObserveCommand.Paused  => SeqAction.fail(SeqexecFailure.Unexpected(s"Observe pause not yet implemented."))
+    }
 
     for {
       id <- dhsFileId(inst)
     } yield Result.Partial(FileIdAllocated(id), doObserve(id).toAction(ActionType.Observe))
   }
 
-  // scalastyle:off
```
Comment (cquiroz) [inline modules/edu.gemini.seqexec.server/src/main/scala/edu/gemini/seqexec/server/SeqTranslate.scala]: 👍

File: modules/edu.gemini.seqexec.server/src/main/scala/edu/gemini/seqexec/server/gmos/GmosEpics.scala [hunk 10]
```diff
@@ -80,8 +80,10 @@ class GmosEpics(epicsService: CaService, tops: Map[String, String]) {
     override protected val cs: Option[CaCommandSender] = Option(epicsService.getCommandSender("gmos::abort"))
   }
 
-  object observeCmd extends EpicsCommand {
-    override protected val cs: Option[CaCommandSender] = Option(epicsService.getCommandSender("gmos::observe"))
+  object observeCmd extends ObserveCommand {
+    private val cs: Option[CaCommandSender] = Option(epicsService.getCommandSender("gmos::observe"))
+    override protected val os: Option[CaApplySender] = Option(epicsService.createObserveSender("gmos::observeCmd",
+      GMOS_TOP + "dc:apply", GMOS_TOP + "dc:observeC", GMOS_TOP + "dc:stop", GMOS_TOP + "dc:abort", ""))
```
Comment (cquiroz) [inline modules/edu.gemini.seqexec.server/src/main/scala/edu/gemini/seqexec/server/gmos/GmosEpics.scala]: String interpolation is your fridn

**Comment by jluhrs** [comment]
I did a test deployment, no problems there.

File: modules/edu.gemini.epics.acm/src/main/java/edu/gemini/epics/acm/CaObserveSenderImpl.java (hunk 1 above)
Comment (jluhrs) [reply modules/edu.gemini.epics.acm/src/main/java/edu/gemini/epics/acm/CaObserveSenderImpl.java]: Well, it has a couple of asserts that should never trigger.

**Pull request merged**